### PR TITLE
Add run-time type checking to loaders

### PR DESCRIPTION
**Pull request by mgramigna**
Modify ValueSetLoader and QuestionnaireLoader to check validity of the resources before returning. Unit tested everything

Apparently the questionnaire we had in our unit tests wasn't actually valid so I changed it.

to test the behavior, feel free to change some of the resources in public/static to be invalid fhir and it should yell at you.

**WEIRDNESS**: For some reason, the valuesets weren't validating against the interfaces because of the description property not playing nicely with the "markdown" data type (??). I think this is a bug in the typescript library, so I added a workaround and will report the issue to that repo

### CODE REVIEW

**Comment by pmadria** [comment]
Reviewed...everything looks good and the functionality is as intended minus the valueset validation.

**Review by mickohanlon23 — COMMENTED** [review]
A little confused about the first test in `QuestionnaireLoader.test.ts`, but I may be missing something. I modified the public/static valuesets and questionnaires, and both appropriately threw errors when not valid.

File: src/__tests__/loaders/QuestionnaireLoader.test.ts [hunk 1]
```diff
@@ -24,3 +24,15 @@ test('correctly query file for a questionnaire', async () => {
 
   expect(actualQuestionnaire).toEqual(sampleQuestionnaire);
 });
+
+test('invalid Questionnaire file should throw error', async () => {
+  expect(() => {
+    questionnaire.getFromFile('./src/__tests__/fixtures/sample-patient-bundle.json');
```
[the remark this answers is not in the document]
Comment (mgramigna) [reply src/__tests__/loaders/QuestionnaireLoader.test.ts:30]: This is intentionally a path to not a questionnaire (a bundle in this case). Ensuring that an improper resources doesn't pass validity.

That's weird that it's still throwing errors on valid things though. I'll look into that

[the remark this answers is not in the document]
Comment (mgramigna) [reply src/__tests__/loaders/QuestionnaireLoader.test.ts:30]: these tests aren't referencing anything in public/static. only the fixtures we have in the tests directory

[the remark this answers is not in the document]
Comment (mickohanlon23) [reply src/__tests__/loaders/QuestionnaireLoader.test.ts:30]: Ah, I see - I wasn't reading it correctly at first.

**Review by mickohanlon23 — APPROVED** [review]
Tested with both the `public/static` questionnaire and valuesets, as well as the test cases in `__tests__/fixtures`. Things work correctly normally, and then throw an error if an integral part of the VS or questionnaire is missing. Looks good!

**Pull request merged**